--- boost/libs/polygon/include/boost/polygon/polygon_45_set_data.hpp	2024-07-31 23:49:08.016048498 +0300
+++ boost/libs/polygon/include/boost/polygon/polygon_45_set_data.hpp.modified	2024-07-31 23:36:20.069293631 +0300
@@ -1578,9 +1578,10 @@
     } else {
       sort();
       rvalue.sort();
-      try {
+      if constexpr (true) {
         result.is_manhattan_ = applyBoolean45OpOnVectors<Unit, op>(result.data_, data_, rvalue.data_);
-      } catch (std::string str) {
+      } else {
+        std::string str;
         std::string msg = "GTL 45 Boolean error, precision insufficient to represent edge intersection coordinate value.";
         if(str == msg) {
           result.clear();
@@ -1659,7 +1660,7 @@
           result.is_manhattan_ = result_is_manhattan;
           result.dirty_ = false;
           result.unsorted_ = false;
-        } else { throw str; }
+        } else { }
       }
       //std::cout << "DONE SCANNING\n";
     }
@@ -1699,9 +1700,10 @@
       result.unsorted_ = false;
     } else {
       sort();
-      try {
+      if constexpr (true) {
         result.is_manhattan_ = applyUnary45OpOnVectors<Unit, op>(result.data_, data_);
-      } catch (std::string str) {
+      } else {
+        std::string str;
         std::string msg = "GTL 45 Boolean error, precision insufficient to represent edge intersection coordinate value.";
         if(str == msg) {
           result.clear();
@@ -1770,7 +1772,7 @@
           result.is_manhattan_ = result_is_manhattan;
           result.dirty_ = false;
           result.unsorted_ = false;
-        } else { throw str; }
+        } else { }
       }
       //std::cout << "DONE SCANNING\n";
     }
